### ISSUE TITLE
QST: formatting for boolean and datetime

### ISSUE THREAD

**Issue by mwx23** [oeg-upm/morph-kgc#67]
I'm using `morph-kgc` to materialise data from a postgres db. 

I've noticed that booleans get extracted as `"True"^^xsd:boolean` with a capital `T` instead of "true"^^xsd:boolean`, lowercase `t`. Same with False/false. According tot the spec http://books.xmlschemata.org/relaxng/ch19-77025.html it should be lowercase.

A similar thing happens with timestamp objects from the db being converted to xsd:dateTime, `"2022-01-01 00:00:00"^^xsd:dateTime` instead of ISO8601 format `"2022-01-01T00:00:00"^^xsd:dateTime`.

I suspect this is something to do with how `pandas` renders things. Is there a mechanism to define a formatter for certain types? 

Thanks for your amazing work on this project!

**Comment by ArenasGuerreroJulian** [comment]
Hi @mwx23 ,

Thanks for using Morph-KGC and reporting the issue!

Indeed, this is something that I knew I just didnt have the time to do it. In the R2RML spec they have it under the section [Natural Mapping of SQL Values](https://www.w3.org/TR/r2rml/#natural-mapping).

This should be easy to fix as for booleans we can convert them to lowercase using [lower](https://pandas.pydata.org/docs/reference/api/pandas.Series.str.lower.html) and for datetime we can use [isoformat](https://pandas.pydata.org/docs/reference/api/pandas.Timestamp.isoformat.html).

I will solve this in the next iteration  .

Julián

**Comment by mwx23** [comment]
Thanks for looking into the issue. In the meantime I just used the following `sed` commands to convert in place.

```bash
# convert True and False xsd:boolean to lowercase true and false
sed -i -e "s/\"True\"\^\^/\"true\"\^\^/" -e "s/\"False\"\^\^/\"false\"\^\^/" triples.nt
# convert datetime to ISO8601 with a T instead of space between date and time
sed -i -re "s/\"([0-9]{4}-[0-9]{2}-[0-9]{2}) ([0-9]{2}:[0-9]{2}:[0-9]{2})(.[0-9]{6})?\"\^\^/\"\1T\2\3\"\^\^/" triples.nt
```

Thanks for all your hard work Julián.

**Comment by ArenasGuerreroJulian** [comment]
Definitely a good solution for the meantime.

Thanks for sharing :)

**Comment by anaigmo** [comment]
Hi! I have a similar issue when loading data to stardog, it outputs error messages for the booleans (as commented above) and for integers. Morph generates for the datatype xsd:integer numbers ending with ".0" as follows:
```
"1611010800.0"^^<http://www.w3.org/2001/XMLSchema#integer>
```
Non negative integers are generated correctly without this issue. In GraphDB or Blazegraph I don't get an error message with this issue though, this triplestore is picky :)

**Comment by ArenasGuerreroJulian** [comment]
This should be fixed now in verion 2.2.0 @mwx23 @anaigmo
